### PR TITLE
feat: quotes list#44

### LINKED ISSUE #44: Quotes: As a B2B Admin User, I should be able to see Quotes list
https://www.figma.com/file/bKJuIwUx6VXmubHZo4rCBq/B2B?type=design&node-id=4-12&mode=design&t=aOy4a7mOltR3zpGJ-0

**// Get Quotes List**
```
query {
  quotes(sortBy: "number") {
    items {
      tenantId
      siteId  
      id
      number
      name
      expirationDate
      submittedDate
      total
      status      
    }
  }
}


```

// sorty by asc order and desc order with following fields
submittedDate 
name
number

**// Get Single Quote**

```
query quote($quoteId: String!, $draft : Boolean) {
  quote(quoteId: $quoteId, draft:$draft){
      tenantId
      siteId         
      number
      name      
      expirationDate
      submittedDate
      total
      status 
     
      items {
        id
        product {
          productCode
          name              
        }
        quantity
        fulfillmentMethod 
        unitPrice {
          listAmount
          saleAmount
        }
        discountTotal
        discountedTotal
        total
        
        
        shippingTotal        
        dutyAmount
        
      }
    
      subTotal        
      shippingSubTotal
      handlingSubTotal
        itemTaxTotal
        shippingTaxTotal
        handlingTaxTotal
        dutyTotal
    
      fulfillmentInfo {
      fulfillmentContact {
        id
        email
        firstName
        lastNameOrSurname
        phoneNumbers {
          home
          mobile
          work
        }
        address {
          address1
          address2
          address3
          address4
          cityOrTown
          stateOrProvince
          postalOrZipCode
          addressType 
          isValidated
        }
      }
    }
         
      comments {
        id
        text
        auditInfo {
          updateDate
          createDate
          updateBy
          createBy
        }
      }
    
      auditHistory {
        changes {
          type
          path
          fields {
            name
            oldValue
            newValue
          }
        }
      }
    
      auditInfo {
        updateDate
        createDate
        updateBy
        createBy
      }

  }
}
```

**// Query variables**
```
{
  "quoteId": "15bfe3f7305f1f00010f3cdb0000933b",
  "draft": true
}
```

**Note:**
Currently updateQuote mutation ignores provided fulfillmentInfo and comments
We have following rest API for the same
  /commerce/quotes/{quoteId}/fulfillmentInfo
  /commerce/quotes/{quoteId}/comments
But corresponding GraphQL mutations are missing.

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: components/b2b/QuotesTable/QuotesTable.spec.tsx [hunk 1]
```diff
@@ -16,50 +19,27 @@ const { Common, Mobile } = composeStories(stories)
...
-      </button>
-      <button data-testid="onStoreSetOrUpdate-button" onClick={onStoreSetOrUpdate}>
-        onStoreSetOrUpdate
-      </button>
-    </div>
-  ),
-}))
+afterEach(() => {
+  cleanup()
+})
```
Comment (kibo-sushant) [inline components/b2b/QuotesTable/QuotesTable.spec.tsx]: I think this is done automatically

File: components/page-templates/B2B/QuotesTemplate/QuotesTemplate.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+describe('[Templates]  QuotesTemplate', () => {
+  it('should render Quotes title', () => {
+    renderWithQueryClient(<Common {...Common.args} />)
+    expect(screen.getByRole('heading', { name: 'quotes', level: 1 })).toBeVisible()
+  })
+
+  it('should render Create Quote button', () => {
+    renderWithQueryClient(<Common {...Common.args} />)
+    expect(screen.getByRole('button', { name: 'create-a-quote' })).toBeVisible()
+  })
```
Comment (kibo-sushant) [inline components/page-templates/B2B/QuotesTemplate/QuotesTemplate.spec.tsx]: should render component
 - test whether title is rendered or not
 - test whether button is rendered or not
Not need of two separate testcases

File: hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  })
+
+  return {
+    data,
+    isLoading,
+    isError,
+    error,
+    isSuccess,
+  }
+}
```
Comment (kibo-sushant) [inline hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts:52]: Add typedoc code

File: hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import type { QueryQuotesArgs } from '@/lib/gql/types'
+
+/**
+ * @hidden
+ */
+
+const client = makeGraphQLClient()
+
+const fetchQuotes = async (param: QueryQuotesArgs) => {
```
Comment (kibo-sushant) [inline hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts]: Add return type

Comment (kibo-chandradeeptalaha) [reply hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts]: Added

File: hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts (hunk 3 above)
Comment (kibo-chandradeeptalaha) [reply hooks/queries/b2b/quotes/useGetQuotes/useGetQuotes.ts:52]: Done

File: components/page-templates/B2B/QuotesTemplate/QuotesTemplate.spec.tsx (hunk 2 above)
Comment (kibo-chandradeeptalaha) [reply components/page-templates/B2B/QuotesTemplate/QuotesTemplate.spec.tsx]: Done

File: components/b2b/QuotesTable/QuotesTable.spec.tsx (hunk 1 above)
Comment (kibo-chandradeeptalaha) [reply components/b2b/QuotesTable/QuotesTable.spec.tsx]: Removed

**Review by kibo-sushant — APPROVED** [review]

**Pull request merged**